### PR TITLE
feat: GSF Debugger

### PR DESCRIPTION
**Pull request by benjaminhuth**
Adds a little debugging tool

Looks like this:

gsf_debugger_0

gsf_debugger_3

### CODE REVIEW

File: Examples/Scripts/GsfDebugger/make_gsf_verbose_log.py
Comment (paulgessinger) [inline Examples/Scripts/GsfDebugger/make_gsf_verbose_log.py:1]: Could this script be inlined  into the test in `test_misc.py`. Does it need to be a separate file?

File: Examples/Scripts/GsfDebugger/src/widgets.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+import matplotlib
+
+matplotlib.use("QtAgg")
+
+from PyQt5 import QtCore, QtWidgets, QtGui
```
Comment (paulgessinger) [inline Examples/Scripts/GsfDebugger/src/widgets.py:5]: This needs Qt? Do you want to supply a reuirements file for this in the `GsfDebugger` folder?

**Review by paulgessinger — COMMENTED** [review]
It seems like a lot of code for debugging to put into the main repo, but I don't have a better suggestion where to put it.

Since the pictures are only for illustration of what the tool looks like (as far as I understand), maybe you can try to compress them (more), or use non-transparent JPGs to reduce the size a bit? It's about 1M of PNGs currently.

File: Examples/Scripts/GsfDebugger/src/widgets.py (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Scripts/GsfDebugger/src/widgets.py:5]: yeah good point, I will provide one

File: Examples/Scripts/GsfDebugger/make_gsf_verbose_log.py
Comment (benjaminhuth) [reply Examples/Scripts/GsfDebugger/make_gsf_verbose_log.py:1]: Hmm how important is this point for you? One advantange of the separate file is that users can try the debugger easily by just creating a logfile with this script...

**Comment by benjaminhuth** [comment]
I have compressed the images to ~500K, and updated the documentation with a bit of text.

**Comment by paulgessinger** [comment]
Merged manually, because this got bumped to the end of the auto merge queue by accident.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [b9416d99613bdfcd567c8ff04892d47f445784b4]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768454)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768453)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768452)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768451)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768450)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768449)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768448)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768447)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768446)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768445)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768444)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768443)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768442)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768441)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768440)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768439)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768438)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768437)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768436)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768435)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768434)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768433)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768432)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768431)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768430)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768429)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768428)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768427)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768426)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768425)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40768424)||

**Pull request merged**